src/NewsService.Business/CreateNewsCommand.cs

```diff
@@ -1,102 +1,131 @@
+          return response.Message.Body.DepartmentIds;
         }
 
-        public CreateNewsCommand(
-            INewsRepository repository,
-            IDbNewsMapper mapper,
-            INewsValidator validator,
-            IAccessValidator accessValidator,
-            IRequestClient<ICheckDepartmentsExistence> rcCheckDepartmentsExistence,
-            ILogger<CreateNewsCommand> logger)
+        _logger.LogWarning("Can not find department with this Id: {departmentId}: " +
+            $"{Environment.NewLine}{string.Join('\n', response.Message.Errors)}");
+      }
+      catch (Exception exc)
+      {
+        _logger.LogError(exc, logMessage);
+      }
+
+      errors.Add(errorMessage);
+      return null;
+    }
+
+    public CreateNewsCommand(
+      INewsRepository repository,
+      IDbNewsMapper mapper,
+      INewsValidator validator,
+      IAccessValidator accessValidator,
+      IRequestClient<ICheckDepartmentsExistence> rcCheckDepartmentsExistence,
+      ILogger<CreateNewsCommand> logger,
+      IHttpContextAccessor httpContextAccessor)
+    {
+      _repository = repository;
+      _mapper = mapper;
+      _validator = validator;
+      _accessValidator = accessValidator;
+      _rcCheckDepartmentsExistence = rcCheckDepartmentsExistence;
+      _logger = logger;
+      _httpContextAccessor = httpContextAccessor;
+    }
+
+    public OperationResultResponse<Guid?> Execute(CreateNewsRequest request)
+    {
+      if (!_accessValidator.HasRights(Rights.AddEditRemoveNews))
+      {
+        _httpContextAccessor.HttpContext.Response.StatusCode = (int)HttpStatusCode.Forbidden;
+
+        return new OperationResultResponse<Guid?>
         {
-            _repository = repository;
-            _mapper = mapper;
-            _validator = validator;
-            _accessValidator = accessValidator;
-            _rcCheckDepartmentsExistence = rcCheckDepartmentsExistence;
-            _logger = logger;
-        }
+          Status = OperationResultStatusType.Failed,
+          Errors = new() { "Not enough rights." }
+        };
+      }
 
-        public OperationResultResponse<Guid> Execute(News request)
+      if (!_validator.ValidateCustom(request, out List<string> errors))
+      {
+        _httpContextAccessor.HttpContext.Response.StatusCode = (int)HttpStatusCode.BadRequest;
+
+        return new OperationResultResponse<Guid?>
         {
-            if (!(_accessValidator.IsAdmin() ||
-                 _accessValidator.HasRights(Rights.AddEditRemoveNews)))
-            {
-                throw new ForbiddenException("Not enough rights.");
-            }
+          Status = OperationResultStatusType.Failed,
+          Errors = errors
+        };
+      }
 
-            _validator.ValidateAndThrowCustom(request);
+      OperationResultResponse<Guid?> response = new();
 
-            OperationResultResponse<Guid> response = new();
+      List<Guid> existDepartments = CheckDepartmentExistence(request.DepartmentId, response.Errors);
```

Comment:
you check only one id, maybe better return bool in this mehtod, and don't send id to mapper and use id from request there

Reply:
Department is not my task, just the rules of the tabs